behaviour/graql/language/define.feature

```diff
@@ -338,66 +744,352 @@ Feature: Graql Define Query
+      | COL | label | colour           |
+      | GRC | label | grayscale-colour |
+    Then uniquely identify answer concepts
+      | x   |
+      | COL |
+      | GRC |
+
+
+  Scenario: define attribute subtype inherits 'key' from all of its supertypes
+    Given graql define
+      """
+      define
+      hex-value sub attribute, value string;
+      colour sub attribute, value string, key hex-value;
+      dark-colour sub colour;
+      dark-red-colour sub dark-colour;
+      very-dark-red-colour sub dark-red-colour;
+      """
+    Given the integrity is validated
+
+    When get answers of graql query
+      """
+      match $x key hex-value; get;
+      """
+    Then concept identifiers are
+      |     | check | value                |
+      | COL | label | colour               |
+      | DRK | label | dark-colour          |
+      | DKR | label | dark-red-colour      |
+      | VDR | label | very-dark-red-colour |
+    Then uniquely identify answer concepts
+      | x   |
+      | COL |
+      | DRK |
+      | DKR |
+      | VDR |
+
+
+  Scenario: define abstract attribute type creates a type
+    Given graql define
+      """
+      define number-of-limbs sub attribute, abstract, value long;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x type number-of-limbs; get;
+      """
+    Then concept identifiers are
+      |     | check | value           |
+      | NOL | label | number-of-limbs |
+    Then uniquely identify answer concepts
+      | x   |
+      | NOL |
+
+
+  Scenario: define concrete subtype of abstract attribute creates child of its parent type
+    Given graql define
+      """
+      define
+      number-of-limbs sub attribute, abstract, value long;
+      number-of-legs sub number-of-limbs;
+      """
+    Given the integrity is validated
+
+    When get answers of graql query
+      """
+      match $x sub number-of-limbs; get;
+      """
+    Then concept identifiers are
+      |     | check | value           |
+      | NOL | label | number-of-limbs |
+      | NLE | label | number-of-legs  |
+    Then uniquely identify answer concepts
+      | x   |
+      | NOL |
+      | NLE |
+
+
+  Scenario: define abstract subtype of abstract attribute creates child of its parent type
+    Given graql define
+      """
+      define
+      number-of-limbs sub attribute, abstract, value long;
+      number-of-artificial-limbs sub number-of-limbs, abstract;
+      """
+    Given the integrity is validated
+
+    When get answers of graql query
+      """
+      match $x sub number-of-limbs; get;
+      """
+    Then concept identifiers are
+      |     | check | value                      |
+      | NOL | label | number-of-limbs            |
+      | NAL | label | number-of-artificial-limbs |
+    Then uniquely identify answer concepts
+      | x   |
+      | NOL |
+      | NAL |
+
+
+  Scenario: define a type as abstract errors if has non-abstract parent types (?)
```

Comment:
this meant to be empty? Along with the ones below?

Reply:
For now, yes. I intend to implement them once #46 is resolved.